src/AppBridgeNative.ts

```diff
@@ -144,47 +148,63 @@ export class AppBridgeNative implements IAppBridgeNative {
     }
 
     public async getColorsByIds(colorIds: number[]): Promise<Color[]> {
-        const response = await window.fetch(`/api/color/library/${window.application.config.context.project.id}`, {
-            method: 'GET',
-            headers: {
-                'x-csrf-token': (document.getElementsByName('x-csrf-token')[0] as HTMLMetaElement).content,
-                'Content-Type': 'application/json',
-            },
-        });
+        const colors: Color[] = await this.getColors();
+
+        return colors.filter((color) => colorIds.includes(Number(color.id)));
+    }
 
-        const responseJson: { palettes: ColorPalette[] } = await response.json();
+    public async getColors(): Promise<Color[]> {
+        const colorPalettes = await this.getColorPalettes();
+
+        let colors: Color[] = [];
+        for (const colorPalette of colorPalettes) {
+            colors = colors.concat(await this.getColorsByColorPaletteId(colorPalette.id));
+        }
```

Comment:
```suggestion
        const colorsPerPalette = await Promise.all(
            colorPalettes.map((colorPalette) => this.getColorsByColorPaletteId(colorPalette.id)),
        );

        return colorsPerPalette.flat(1);
```
Should allow parallel API calls